src/hmem.c

```diff
@@ -404,17 +404,19 @@ static ssize_t ofi_copy_mr_iov(struct ofi_mr **mr, const struct iovec *iov,
 			hmem_data = NULL;
 		}
 
-		if (hmem_iface == FI_HMEM_CUDA && (hmem_flags & OFI_HMEM_DATA_GDRCOPY_HANDLE)) {
+		if (hmem_flags & OFI_HMEM_DATA_DEV_REG_HANDLE) {
 			/**
-			 * TODO: Fine tune the max data size to switch from gdrcopy to cudaMemcpy
-			 * Note: buf must be on the host since gdrcopy does not support D2D copy
+			 * TODO: Fine tune the max data size to switch from
+			 * def_reg copy to normal copy
```

Comment:
I wasn't clear in my previous comment. I think we should remove this comment in src/hmem.c

This is the common code path and not concerned with tuning(which is up to the provider).